Homework_1/report/main.tex

```diff
@@ -0,0 +1,141 @@
+
+\usepackage{color}
+\usepackage{graphicx} % slike
+\usepackage{caption}
+\usepackage{subcaption}
+\usepackage{siunitx} % enote
+\usepackage{amsmath} % enačbe
+\usepackage{amssymb} % simboli
+\usepackage[thinc]{esdiff} % za odvode
+\usepackage[colorlinks=true]{hyperref} % za linke
+\usepackage{float} % za fiksiranje figur
+\usepackage{mathtools}
+\usepackage{esint} % za fancy integrale
+\usepackage[numbib,nottoc]{tocbibind}
+\usepackage[slovene]{babel}
+
+
+\usepackage{tikz}
+
+\newcommand{\dif}[1]{\,{\rm d}#1}
+
+\begin{document}
+
+\begin{titlepage}
+    \begin{center}
+        \includegraphics[width=0.5\textwidth]{figures/FRI_logo.png}\\
+        \vspace{0.5cm}
+        \vspace{3cm}
+        {\LARGE \bf QR razcep simetrične tridiagonalne matrike} \\
+        \vspace{0.3cm}
+        \vspace{2.0cm}
+        {\large Numerična matematika}\\
+        \vspace{0.2cm}
+        {|}\\
+        \vspace{0.2cm}
+        {\large 1. domača naloga}\\
+        \vspace{2.0cm}
+    \end{center}
+    \vfill
+    \begin{flushleft}
+        {\normalsize {\sf Avtor:} Vito Levstik\\}
+    \end{flushleft}
+    \vspace{2cm}
+    \begin{center}
+        {\normalsize \sc Akademsko leto 2024/2025}
+    \end{center}
+\end{titlepage}
+
+\newpage
+
+\section{Uvod}
+
+Cilj te naloge je implementirati učinkovit QR razcep (simetrične) tridiagonalne matrike z Givensovimi rotacijami 
+in ga uporabiti pri QR iteraciji za izračun lastnih vrednosti in lastnih vektorjev simetrične tridiagonalne matrike.
+
+Da lahko to dosežemo, je najprej potrebno implementirati tipe \texttt{Tridiag}, ki predstavlja tridiagonalno matriko,
+\texttt{SimTridiag}, ki predstavlja simetrično tridiagonalno matriko, \texttt{Givens}, ki predstavlja matriko Givensovih rotacij in
+\texttt{ZgornjeTridiag}, ki predstavlja matriko z neničelnimi elementi na diagonali in dveh zgornjih diagonalah.
+
+Glavna naloga je tako implementacija funkcije \texttt{qr($T$)}, ki tridiagonalno matriko $T$ razcepi v produkt matrik $Q$ in $R$, kjer je $Q$ tipa \texttt{Givens} in $R$ tipa \texttt{ZgornjeTridiag} in
+implementacija funkcije \texttt{eigen($T$)}, ki izračuna lastne vrednosti in lastne vektorje simetrične tridiagonalne matrike $T$ z uporabo QR iteracije.
+
+\section{Implementacija}
+Podatkovni tip \texttt{Tridiag} sprejme tri sezname in sicer seznam spodnje diagonale, seznam glavne diagonale in seznam zgornje diagonale.
+Tale podatkovni tip smo že implementirali na vajah, zato je koda povzeta iz vaj. Podatkovni tip \texttt{SimTridiag} je podobno definiran, le da sprejme dva seznama, seznam glavne diagonale in seznam pod-diagonale, 
+saj sta zgornja in spodnja diagonala enaka. Podatkovni tip \texttt{ZgornjeTridiag} je podoben, le da sprejme seznam glavne diagonale in dva seznama zgornjih diagonal.
+
+Do elementov vseh treh matrik lahko dostopamo z indeksi, kot pri običajni matriki, kar smo storili z implementacijo funkcij \texttt{getindex()}. Elemente matrik lahko tudi spreminjamo s funkcijo \texttt{setindex()},
```

Comment:
Odločite se ali boste pisali v sedanjiku, prihodnjiku ali pretekliku in isti čas uporabite povsod. Ni boljšega časa kot sedanjost, zato predlagam, da povsod uporabljate sedanjik.